webpack/assets/javascripts/react_app/components/AutoCompleteSearch/AutoComplete.js

```diff
@@ -0,0 +1,98 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import debounce from 'lodash/debounce';
+import { Icon, TypeAheadSelect } from 'patternfly-react';
+import { STATUS } from '../../constants';
+import helpers from '../../common/helpers';
+import './auto-complete.scss';
+
+class AutoComplete extends React.Component {
+  constructor(props) {
+    super(props);
+    props.getInitialResults(props.data);
+    this.handleInputChange = debounce(this.handleInputChange, 250);
+    this.handleInputFocus = debounce(this.handleInputFocus, 250);
```

Comment:
I think the problem with focus is that it would fetch the same data many times,
and what we really need is getting the basic results once the page is loaded, and than update the results only on input changes.

Reply:
so just query it once, if it was in focus and its in the right state make the api call?